great_expectations/data_context/data_context/cloud_data_context.py

```diff
@@ -919,3 +921,26 @@ def _view_validation_result(self, result: CheckpointResult) -> None:
             url
         ), "Guaranteed to have a validation_result_url if generating a CheckpointResult in a Cloud-backed environment"
         self._open_url_in_browser(url)
+
+    def _add_datasource(
+        self,
+        name: str | None = None,
+        initialize: bool = True,
+        save_changes: bool | None = None,
+        datasource: BaseDatasource | FluentDatasource | LegacyDatasource | None = None,
+        **kwargs,
+    ) -> BaseDatasource | FluentDatasource | LegacyDatasource | None:
+        result = super()._add_datasource(
+            name=name,
+            initialize=initialize,
+            save_changes=save_changes,
+            datasource=datasource,
+        )
+        if result and not isinstance(result, FluentDatasource):
+            # deprecated-v0.17.2
+            warnings.warn(
+                "Adding block-style or legacy datasources in a Cloud-backed environment is deprecated as of v0.17.2 and will be removed in v0.19. "
```

Comment:
Can we speed up this process since this is only for for Cloud?
I don't think we have to abide by the same OSS deprecations processes.
```suggestion
                "Adding block-style or legacy datasources in a Cloud-backed environment is deprecated as of v0.17.2 and will be removed in a future version. "
```

Reply:
yup I agree